compiler/src/iree/compiler/Dialect/Flow/Transforms/CollapseDimensions.cpp

```diff
@@ -200,15 +227,6 @@ findRootGenericOp(DispatchRegionOp regionOp) {
     }
   }
 
-  // Check that the operands of the generic op are defined outside the dispatch.
```

Comment:
[no action required]:  I don't understand why we had the check. But the changes that you made look reasonable to me.